host/class/uac/usb_host_uac/include/usb/uac_host.h

```diff
@@ -0,0 +1,374 @@
+*/
+typedef struct {
+    bool create_background_task;            /*!< When set to true, background task handling USB events is created.
+                                             Otherwise user has to periodically call uac_host_handle_events function */
+    size_t task_priority;                   /*!< Task priority of created background task */
+    size_t stack_size;                      /*!< Stack size of created background task */
+    BaseType_t core_id;                     /*!< Select core on which background task will run or tskNO_AFFINITY  */
+    uac_host_driver_event_cb_t callback;    /*!< Callback invoked when UAC driver event occurs. Must not be NULL. */
+    void *callback_arg;                     /*!< User provided argument passed to callback */
+} uac_host_driver_config_t;
+
+/**
+ * @brief UAC logic device/interface configuration structure
+ *
+*/
+typedef struct {
+    uint8_t addr;                                       /*!< USB Address of connected physical device */
+    uint8_t iface_num;                                  /*!< UAC Interface Number */
+    uint32_t buffer_size;                               /*!< Audio buffer size */
+    uint32_t buffer_threshold;                          /*!< Audio buffer threshold */
+    uac_host_device_event_cb_t callback;                /*!< Callback invoked when UAC device event occurs */
+    void *callback_arg;                                 /*!< User provided argument passed to callback */
+} uac_host_device_config_t;
+
+/**
+ * @brief UAC stream configuration structure
+ *
+*/
+typedef struct {
+    uint8_t channels;                                    /*!< Audio channel number */
+    uint8_t bit_resolution;                              /*!< Audio bit resolution */
+    uint32_t sample_freq;                                /*!< Audio sample resolution */
+    uint16_t flags;                                      /*!< Control flags */
+} uac_host_stream_config_t;
+
+/**
+ * @brief UVC control type
+*/
+typedef enum {
+    UAC_CTRL_NONE = 0,        /*!< None */
+    UAC_CTRL_UAC_MUTE,        /*!< mute control. ctrl_data (false/true) */
+    UAC_CTRL_UAC_VOLUME,      /*!< volume control. ctrl_data (0~100) */
+    UAC_CTRL_UAC_VOLUME_DB,   /*!< volume control. ctrl_data DB value */
+    UAC_CTRL_MAX,             /*!< max type value */
+} uac_host_ctrl_t;
+
+// ----------------------------- Public ---------------------------------------
+/**
+ * @brief Install USB Host UAC Class driver
+ *
+ * @note This function must be called after usb_host_install
+ *
+ * @param[in] config UAC driver configuration structure
+ * @return esp_err_r
+ *  - ESP_OK on success
+ *  - ESP_ERR_INVALID_STATE if UAC driver is already installed
+ *  - ESP_ERR_INVALID_ARG if the configuration is invalid
+ *  - ESP_ERR_NO_MEM if memory allocation failed
+ *
+*/
+esp_err_t uac_host_install(const uac_host_driver_config_t *config);
+
+/**
+ * @brief Uninstall USB Host UAC Class driver
+ *
```

Comment:
Can you add some note about closing device before calling this function? Something like:

```suggestion
 *
 * note This function must be called after a device was closed
```